### PR TITLE
jsdom: add ResourceLoader

### PR DESCRIPTION
**Pull request by joscha**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/jsdom/jsdom/pull/2279 and https://github.com/jsdom/jsdom/pull/2339
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/jsdom/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for jsdom 11.12
+// Type definitions for jsdom 12.2
```
Comment (joscha) [inline types/jsdom/index.d.ts]: RresourceLoader was added in 12.0, the `element` option (see second linked PR) in 12.2

File: types/jsdom/index.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 // Project: https://github.com/tmpvar/jsdom#readme
 // Definitions by: Leonard Thieu <https://github.com/leonard-thieu>
 //                 Johan Palmfjord <https://github.com/palmfjord>
```
Comment (joscha) [inline types/jsdom/index.d.ts]: can add myself here if you guys @leonard-thieu & @palmfjord don't want to own the ResourceLoader part

Comment (palmfjord) [reply types/jsdom/index.d.ts]: Sure, feel free!

**Review by palmenhq — APPROVED** [review]
Lgtm at least 🙌

**Pull request merged**